ksqldb-engine/src/main/java/io/confluent/ksql/util/SharedKafkaStreamsRuntimeImpl.java

```diff
@@ -127,7 +138,16 @@ public synchronized void close() {
   public void start(final QueryId queryId) {
     if (collocatedQueries.containsKey(queryId) && !collocatedQueries.get(queryId).everStarted) {
       if (!kafkaStreams.getTopologyByName(queryId.toString()).isPresent()) {
-        kafkaStreams.addNamedTopology(collocatedQueries.get(queryId).getTopology());
+        try {
+          kafkaStreams.addNamedTopology(collocatedQueries.get(queryId).getTopology())
+              .all()
+              .get(30, TimeUnit.SECONDS);
+        } catch (final TimeoutException | ExecutionException | InterruptedException e) {
+          e.printStackTrace();
```

Comment:
Ditto here -- please log an actual error message instead of just a hanging stacktrace.